components/brave_wallet/browser/bitcoin/bitcoin_discover_account_task.cc

```diff
@@ -73,17 +67,20 @@ DiscoverAccountTaskBase::DiscoverAccountTaskBase(
 }
 DiscoverAccountTaskBase::~DiscoverAccountTaskBase() = default;
 
+DiscoverAccountTaskBase::State& DiscoverAccountTaskBase::GetState(
+    DiscoveryStage stage) {
+  return stage == DiscoveryStage::kReceive ? receive_addresses_state_
+                                           : change_addresses_state_;
```

Comment:
Only two. Ternary operator seems fine there.

Reply:
Because this is an enum, we should use a switch here then, because that will make the code more correct in case of changes, and more readable to the next reader. It is better to treat enums as values, than as if it was a boolean (e.g. `is_receive_state`)

```cxx
switch (stage) {
  case DiscoveryStage::kReceive:
    return receive_addresses_state_;
  case DiscoveryStage::kChange:
    return change_addresses_state_;
}
NOTREACHED();
```
The switch here is more readable, because it provides all the context necessary to the reader. It is also more correct, as we are using the compiler to validate that we are covering all the entries of the enum.

Reply:
> Because this is an enum, we should use a switch here then, because that will make the code more correct in case of changes, and more readable to the next reader.

Where this "should" comes from? This enum is never going to change and better readability is arguable.
Chromium has a lot of examples like these:
https://source.chromium.org/chromium/chromium/src/+/main:components/signin/public/base/signin_metrics.cc;l=340
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/dips/dips_utils.h;l=49

I'll change it to a bool.

Reply:
You don't need to change to a bool. An enum is more descriptive. Keep as an enum and just add the switch please.

Reply:
bool is descriptive enough

Reply:
Like I said on the original request, on the first PR, this should be a switch.

Reply:
Again, what are your arguments for "should"?

Reply:
I feel we are getting dragged here into something that is much smaller than it seems. I've explained above already why I would prefer a switch, but I don't think we need to get stuck on this.